### PR TITLE
Feature/support collectible history

### LINKED ISSUE #2330: [Collectibles] History row + details
Let's port this from mobile.

This task is about mapping and displaying the "Sent collectible" history row + details.

[Figma](https://www.figma.com/design/C3G0a4Gd6RQyplRBppGDsL/Freighter-Extension?node-id=6302-30288&t=i6DQHvjsIvsFy5wX-1):
* We have designs for the [history details screen on the extension](https://www.figma.com/design/C3G0a4Gd6RQyplRBppGDsL/Freighter-Extension?node-id=6302-30288&t=i6DQHvjsIvsFy5wX-1) but we are [still missing designs for the history row](https://www.figma.com/design/C3G0a4Gd6RQyplRBppGDsL?node-id=6302-28585#1484303648) so I'm attaching below the [mobile design for the history row](https://www.figma.com/design/KwkHXQxbNmDllwermJtnRu/Freighter-Mobile?node-id=6597-125427&t=jtTZA9y1VVeP4c3V-1) as reference.

Image

Image

### PR DESCRIPTION
**Pull request by piyalbasu**
Closes #2330 & #2472

Figma: https://www.figma.com/design/KwkHXQxbNmDllwermJtnRu/Freighter-Mobile?node-id=8688-92272&m=dev

# Add Collectible History Support 

## Overview

This PR adds comprehensive support for displaying collectible (NFT) transactions in the account history view. Users can now see collectible transfers in their transaction history with proper metadata, images, and collection information.

### 🎨 Collectible History Support
- **Collectible Transaction Display**: Collectible transfers (SEP-50) are now properly detected and displayed in account history
- **Rich Metadata**: Shows collectible name, collection name, token ID, and images in history rows
- **Transaction Detail Modal**: Enhanced modal displays collectible information with images and metadata
- **Batch Fetching**: Implements efficient batch fetching to avoid N+1 query problems when loading multiple collectibles

### 🔧 Technical Improvements
- **Batch Collectible Fetcher**: New utility for efficiently fetching multiple collectibles in a single request
- **Token List Integration**: Checks token lists for asset icons as fallback

### New Utilities
- `collectiblesBatchFetcher.ts`: Batch fetches collectibles for multiple contracts (needed for history view)
- `collectiblesCache.ts`: Manages caching of collectible collections
- `imagePreloader.ts`: Preloads collectible images for better performance
- `collectibles.ts`: Helper functions for collectible operations

### Enhanced Components
- `TransactionDetail`: Refactored to support collectible transfers rendering
- `HistoryItem`: Updated to display collectible-specific information

### API Changes
- Added `getCollectibles` helper function in `@shared/api/helpers`
- Enhanced `getAttrsFromSorobanHorizonOp` to detect collectible transfers

## Bug Fixes

- ✅ Fixes missing Soorban icons for transfers in history
- ✅ Makes sure to consult with token list before fetching icon from issuer for assets in History (needed for assets the user doesn't already own)
- ✅ Fixes incorrect z-index for icons in History

### Unit Tests
- ✅ `collectiblesBatchFetcher.test.ts` (355 lines) - Comprehensive batch fetching tests
- ✅ `collectiblesCache.test.ts` (314 lines) - Cache management tests
- ✅ `imagePreloader.test.ts` (373 lines) - Image preloading tests
- ✅ `collectibles.test.ts` (250 lines) - Collectible helper function tests
- ✅ `formatters.test.ts` (55 lines) - Formatter utility tests
- ✅ `soroban.test.js` (158 lines) - Enhanced Soroban helper tests
- ✅ `AccountHistory.test.tsx` (578 lines added) - History view integration tests

### E2E Tests
- ✅ `accountHistory.test.ts` (315 lines added) - End-to-end tests for collectible history and asset diffs

### Test Coverage
- Comprehensive coverage for all new utilities and helpers
- Integration tests for collectible history display
- Edge case handling (missing metadata, failed fetches, etc.)

### Key Files
- `extension/src/popup/views/AccountHistory/hooks/useGetHistoryData.tsx` - Main history data processing
- `extension/src/popup/components/accountHistory/TransactionDetail/index.tsx` - Transaction detail modal
- `extension/src/helpers/utils/collectibles/` - New collectible utilities
- `extension/src/popup/helpers/soroban.ts` - Enhanced Soroban operation parsing
- `extension/e2e-tests/accountHistory.test.ts` - E2E test coverage

## Migration Notes

- No breaking changes to existing APIs
- Backward compatible with existing transaction history
- Collectible history gracefully falls back to generic invocation if metadata is unavailable

## Related Issues

- Implements collectible history support requested in collectibles feature
- Addresses performance concerns with batch fetching

## Screenshots

With found metadata:
Screenshot 2026-01-12 at 9 18 09 PM
Screenshot 2026-01-12 at 9 18 17 PM

With missing metadata:
Screenshot 2026-01-12 at 9 18 45 PM
Screenshot 2026-01-12 at 9 18 54 PM

## Checklist

- [x] Code follows project style guidelines
- [x] All tests pass
- [x] No console.log statements in production code
- [x] Error handling uses captureException
- [x] TypeScript types are properly defined
- [x] No linting errors
- [x] Documentation updated where necessary

### CODE REVIEW

File: @shared/api/helpers/fetchCollectibles.ts [hunk 1]
```diff
@@ -118,19 +118,15 @@ export const fetchCollectibles = async ({
     for (let i = 0; i < data.collections.length; i++) {
       const { collection } = data.collections[i];
       if (collection) {
-        const filteredCollectibles = collection.collectibles.filter(
-          (collectible) => collectible.owner === publicKey,
-        );
-
-        if (filteredCollectibles.length > 0) {
```
Comment (piyalbasu) [inline @shared/api/helpers/fetchCollectibles.ts:125]: removing this check for `owner` because for history, we will need to return collectibles that may not belong to the user anymore

File: @shared/constants/soroban/token.ts [hunk 2]
```diff
@@ -4,14 +4,19 @@ export enum SorobanTokenInterface {
   mint = "mint",
 }
 
-export type ArgsForTokenInvocation = {
+export enum SorbanCollectibleInterface {
+  transfer = "transfer",
```
Comment (piyalbasu) [inline @shared/constants/soroban/token.ts:8]: this has the same name as `SorobanTokenInterface`, but for code clarity, I think it's better that it has its own enum

File: extension/src/helpers/hooks/useGetCollectibles.tsx [hunk 3]
```diff
@@ -1,121 +1,152 @@
...
 
 import { initialState, reducer } from "helpers/request";
+import { State } from "constants/request";
 import { Collectibles } from "@shared/api/types/types";
 import { fetchCollectibles } from "@shared/api/helpers/fetchCollectibles";
 import { NetworkDetails } from "@shared/constants/stellar";
 import { captureException } from "@sentry/browser";
 import { getCollectibles } from "@shared/api/internal";
 import { AppDispatch } from "popup/App";
-import { saveCollections, collectionsSelector } from "popup/ducks/cache";
```
Comment (piyalbasu) [inline extension/src/helpers/hooks/useGetCollectibles.tsx:11]: in this file, we haven't changed functionality. We're just cleaning up the code a bit by moving the image preloader and cache checks to helper files

File: extension/src/popup/components/SlideupModal/styles.scss [hunk 4]
```diff
@@ -6,7 +6,7 @@
   bottom: 0;
   left: 0;
   width: 100%;
-  z-index: var(--back--button-z-index);
+  z-index: var(--z-index-modal);
```
Comment (piyalbasu) [inline extension/src/popup/components/SlideupModal/styles.scss:9]: Fixes icon z-index issue

File: extension/src/popup/components/account/AccountCollectibles/index.tsx [hunk 5]
```diff
@@ -9,101 +10,96 @@ import {
...
 import "./styles.scss";
 import { CollectibleInfoImage } from "../CollectibleInfo";
 
 const CollectionsList = ({ collections }: { collections: Collection[] }) => {
-  const { t } = useTranslation();
   const [selectedCollectible, setSelectedCollectible] =
     useState<SelectedCollectible | null>(null);
 
-  // every collection has an error, so nothing to render
-  if (collections.every((collection) => collection.error)) {
```
Comment (piyalbasu) [inline extension/src/popup/components/account/AccountCollectibles/index.tsx:23]: this are filtered out by `getUserCollections` so no need to do this explicitly

File: extension/src/popup/views/AccountHistory/hooks/useGetHistoryData.tsx [hunk 6]
```diff
@@ -163,7 +184,18 @@ export const getTransferIcons = ({
...
+          <AssetSds
+            size="lg"
+            variant="single"
+            sourceOne={{
+              altText: i18n.t("Token logo"),
+              image: icon,
+            }}
+          />
+        ) : (
+          <IconSoroban />
```
Comment (piyalbasu) [inline extension/src/popup/views/AccountHistory/hooks/useGetHistoryData.tsx:197]: small semi-related fix here: we weren't correctly showing the Soroban icon for `transfer` invocations in History

File: extension/src/popup/views/AccountHistory/hooks/useGetHistoryData.tsx [hunk 7]
```diff
@@ -309,22 +356,35 @@ const getIconUrl = async ({
...
-    });
+    if (cachedTokenLists.length > 0) {
+      const { icon } = await getIconFromTokenLists({
+        issuerId: key,
+        code,
+        assetsListsData: cachedTokenLists,
+      });
+      if (icon) {
+        iconUrl = icon;
+      }
```
Comment (piyalbasu) [inline extension/src/popup/views/AccountHistory/hooks/useGetHistoryData.tsx:378]: related to the above fix for Soroban token icons: make sure to check a token list for an image before trying to fetch icon url from the issuer (for classic assets)

File: @shared/api/helpers/fetchCollectibles.ts (hunk 1 above)
Comment (aristidesstaffieri) [reply @shared/api/helpers/fetchCollectibles.ts:125]: it looks like this helper is still used in the account data hook, does that not need this filter still?

Comment (aristidesstaffieri) [reply @shared/api/helpers/fetchCollectibles.ts:125]: oh nevermind, I read far enough to see the related change now

File: @shared/constants/soroban/token.ts [hunk 8]
```diff
@@ -4,14 +4,19 @@ export enum SorobanTokenInterface {
   mint = "mint",
 }
 
-export type ArgsForTokenInvocation = {
+export enum SorbanCollectibleInterface {
```
Comment (aristidesstaffieri) [inline @shared/constants/soroban/token.ts]: typo `Sorban` -> `Soroban`

File: extension/src/helpers/utils/collectibles/collectiblesBatchFetcher.ts [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+    return { collections: [], fromCache: true };
+  }
+
+  // Check cache first if enabled
+  if (useCache) {
+    const cached = getCachedCollections(networkDetails.network, publicKey);
+
+    if (hasValidCache(cached)) {
+      // Check if all requested contracts are in cache
+      const allContractsCached = contracts.every((contract) =>
```
Comment (aristidesstaffieri) [inline extension/src/helpers/utils/collectibles/collectiblesBatchFetcher.ts]: what about cases where the contract is cached but not all token IDs that you own are cached? At a glance, it seems like this should account for token IDs as well.

**Comment by piyalbasu** [comment]
@copilot open a new pull request to apply changes based on the comments in [this thread](https://github.com/stellar/freighter/pull/2505#pullrequestreview-3662235723)

File: extension/src/helpers/utils/collectibles/collectiblesBatchFetcher.ts (hunk 9 above)
Comment (piyalbasu) [reply extension/src/helpers/utils/collectibles/collectiblesBatchFetcher.ts]: Ah yes, very good catch - that is addressed here: https://github.com/stellar/freighter/pull/2505/commits/2bd7e6feba102a4f2bf3f593c51f51c9459e748b
